File: src/Image4ioSDK/TransformationClient.vb
Public Class TransformationClient
    Implements ITransformation


    Private Property ImageMetadata As JSON.JSON_ImageMetadata

    Sub New(ImageMetadata As JSON.JSON_ImageMetadata)
        Me.ImageMetadata = ImageMetadata
    End Sub


#Region "ToWebP"
    Public Function _ToWebP(Quality As Integer, Optional Width As Integer? = Nothing, Optional Height As Integer? = Nothing) As String Implements ITransformation.ToWebP
        Dim sb As New Text.StringBuilder
        sb.AppendFormat("f_webp")
        sb.AppendFormat(",q_{0}", Quality)
        If Width.HasValue Then sb.Append(",w_").Append(Width.Value)
        If Height.HasValue Then sb.Append(",h_").Append(Height.Value)

        Return String.Format("https://cdn.image4.io/{0}/{1}{2}", CloudName, sb.ToString(), ImageMetadata.Path)
    End Function
#End Region

#Region "Resize"
    Public Function _Resize(Quality As Integer, Optional Width As Integer? = Nothing, Optional Height As Integer? = Nothing) As String Implements ITransformation.Resize
        Dim sb As New Text.StringBuilder
        sb.AppendFormat("q_{0}", Quality)
        If Width.HasValue Then sb.Append(",w_").Append(Width.Value)
        If Height.HasValue Then sb.Append(",h_").Append(Height.Value)

        Return String.Format("https://cdn.image4.io/{0}/{1}{2}", CloudName, sb.ToString(), ImageMetadata.Path)
    End Function
#End Region

#Region "Compress"
    Public Function _Compress(Quality As Integer) As String Implements ITransformation.Compress
        Return _Resize(Quality)
    End Function
#End Region


End Class

File: Image4ioSDK/Image4ioSDK/TransformationClient.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Image4ioSDK
{
    public class TransformationClient : ITransformation
    {

        private JSON.JSON_ImageMetadata ImageMetadata;
        public TransformationClient(JSON.JSON_ImageMetadata ImageMetadata)
        {
            this.ImageMetadata = ImageMetadata;
        }


        #region ToWebP
        public string ToWebP(int Quality, int? Width = null, int? Height = null)
        {
            StringBuilder sb = new StringBuilder();
            sb.AppendFormat("f_webp");
            sb.AppendFormat(",q_{0}", Quality);
            if (Width.HasValue) sb.Append(",w_").Append(Width.Value);
            if (Height.HasValue) sb.Append(",h_").Append(Height.Value);

            return $"https://cdn.image4.io/{Basic.CloudName}/{sb.ToString()}{ImageMetadata.Path}";
        }
        #endregion

        #region Resize
        public string Resize(int Quality, int? Width = null, int? Height = null)
        {
            StringBuilder sb = new StringBuilder();
            sb.AppendFormat("q_{0}", Quality);
            if (Width.HasValue) sb.Append(",w_").Append(Width.Value);
            if (Height.HasValue) sb.Append(",h_").Append(Height.Value);

            return $"https://cdn.image4.io/{Basic.CloudName}/{sb.ToString()}{ImageMetadata.Path}";
        }
        #endregion

        #region Compress
        public string Compress(int Quality)
        {
            return Resize(Quality);
        }
        #endregion


    }
}
